### PR TITLE
#CU-2jvu8rt rust tooling for chains api regenerated

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
Signed-off-by: Dzmitry Lahoda <dzmitry@lahoda.pro>

## Description
- made port of rpc of devnet chains more stable (need to to discover API)
- moved generator code to separate package
- made generator generic
- added tooling to run it in ci
- fixed picasso nix
- upgraded subxt

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2jvu8rt Opening XCMP channels: Picasso - Bifrost](https://app.clickup.com/t/2jvu8rt)

File: .config/dictionaries/technical.dic [hunk 1]
```diff
@@ -66,6 +66,8 @@ bignumbers
 binop
 bintools
 bitfield
+Bitfields
+bitfields
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:70]:
```suggestion
bitfields
```
Nit letter case. Cspell currently isn't case-sensitive.

File: .config/dictionaries/technical.dic [hunk 2]
```diff
@@ -170,6 +173,9 @@ miniserve
 minvested
 mochawesome
 msapplication
+Multiaddr
+Multiassets
+multiassets
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:178]:
```suggestion
multiassets
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 3]
```diff
@@ -210,6 +219,9 @@ peekable
 peerset
 perbill
 permill
+
+permissionlessly
+Permissionlessly
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:224]:
```suggestion
permissionlessly
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 4]
```diff
@@ -224,15 +236,25 @@ popout
 powd
 powf
 prebaked
+preclaim
+Preclaim
+preclaims
+Preclaims
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:242]:
```suggestion
preclaim
preclaims
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 5]
```diff
@@ -224,15 +236,25 @@ popout
...
 precommit
 precommits
+Prevote
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:245]:
```suggestion
prevote
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 6]
```diff
@@ -224,15 +236,25 @@ popout
...
 proptests
 prost
 protoc
 PROTOC
 PTRACE
+Rebag
+rebag
+Rebagged
+Rebond
+rebond
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:257]:
```suggestion
rebag
rebagged
rebond
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 7]
```diff
@@ -305,9 +327,21 @@ tungstenite
...
 typeorm
 udeps
+Unbond
+unbond
+Unbonded
+unbonded
+unbonding
+Unbonding
+Unfound
+unfound
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:337]:
```suggestion
unbond
unbonded
unbonding
unfound
```
Nit letter case

File: .config/dictionaries/technical.dic [hunk 8]
```diff
@@ -305,9 +327,21 @@ tungstenite
...
 unlookup
+unpruned
 unstakeform
 unsubs
+Unvouch
+unvouch
+unvouched
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:344]:
```suggestion
unvouch
unvouched
```
Nit letter case

**Review by Martin-Kiss — APPROVED** [review]

File: .config/dictionaries/technical.dic [hunk 9]
```diff
@@ -170,6 +172,8 @@ miniserve
 minvested
 mochawesome
 msapplication
+Multiaddr
```
Comment (Martin-Kiss) [inline .config/dictionaries/technical.dic:175]:
```suggestion
multiaddr
```
Nit, my bad missed it before.

**Review by Martin-Kiss — APPROVED** [review]

File: code/utils/composable-subxt/src/bin/codegen.rs [hunk 10]
```diff
@@ -0,0 +1,63 @@
+//! Can be run withing `cargo run` and `cargo build`.
```
Comment (Dom-Roth) [inline code/utils/composable-subxt/src/bin/codegen.rs:1]:
```suggestion
//! Can be run within `cargo run` and `cargo build`.
```
nit: typo

**Review by Dom-Roth — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Comment by dzmitry-lahoda** [comment]
merged as paft of other pr buikt on top

**Pull request closed without merging**